services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -233,28 +235,43 @@ public boolean checkTokenValidForIntrospection(KeycloakSession session, RealmMod
             return false;
         }
 
-        boolean valid = false;
+        TokenRevocationStoreProvider revocationStore = session.getProvider(TokenRevocationStoreProvider.class);
+        if (revocationStore.isRevoked(token.getId())) {
```

Comment:
Just to confirm my understanding.

Now that token revocation supports access tokens, we have this check to revoke them although in case the token is stateful the session will still be alive. So clients would be forced to refresh the token.

Makes sense ?

Reply:
Yes, the session will be still alive once access tokens are revoked. Few points to this:
-  Specification https://tools.ietf.org/html/rfc7009 does not prescribe the behaviour in this case. It explicitly mentions: "If the token passed to the request is an access token, the server MAY revoke the respective refresh token as well." .

- Regarding session revoke for access tokens, I was thinking about the following scenario:
-- Public client "foo" obtains tokens through the Browser UI
-- Client "foo" sends the token to the resource-server "bar"
-- Resource server "bar" is bad citizen and he decides to revoke the access token sent to it. Nothing prevents him to do so, as the "foo" is public client and hence Keycloak will just need to accept the access token sent to it for the revocation
-- Now if the session will be logged-out as well, it would give the resource server "bar" the power to revoke sessions of client "foo" . Which IMO is not great behaviour. That's why I've not invalidated the session for the access token.

WDYT?

Reply:
The main reason behind my question is that accordingly to the specs the AS "verifies whether the token
   was issued to the client making the revocation request". That said, you can't revoke the token using a different client. In your case, the RS won't be able to revoke tokens issued to the client application.

I did not consider this point as critical because I consider this as an improvement so that admins can choose the behavior.

Reply:
Hmm... The thing is that when "foo" is public client as in my example, then the resource server can just do revocation request? Reason is that there is no client authentication. So the resource server will just need to know:
- the "client_id=foo", which he can use in the revocation request
- the access token, which was sent to it
And he has all of this, so Keycloak server will just need to accept revocation request and revoke the token?

Reply:
I'm not sure if that would be possible because token revocation requires client authentication.

Reply:
Yeah, however in case of the public clients, the client authentication is just verification that there is parameter like "client_id=foo" . It seems our TokenRevocationEndpoint calls the "AuthorizeClientUtil.authorizeClient", which does just this for public clients. So it is not real authentication. The specs does not enforce client authentication, it allows public clients as well, so we are fine with regards to the specs IMO...

Reply:
I see new. Missed the client-id based authentication for public clients.

We should be fine, yes, but your concern is still relevant. Something we can keep an eye on and improve later.